From afea9c53c26146eb7a11e49dd7acf1968bbc59a7 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Wed, 20 Nov 2024 20:59:28 +0000
Subject: [PATCH] wasm: Enable filesystem APIs in libcxx

As of a recent fix included in LLVM 17[1] and wasi-libc fix[2], we can
enable `LIBCXX_ENABLE_FILESYSTEM` in libcxx build for WebAssembly/WASI.
This allows us to use `<filesystem>`, `<fstream>`, etc in C++ code.

[1]: https://github.com/llvm/llvm-project/commit/66a562d22e708ba40b8443b58e504ac3f983ba59
[2]: https://github.com/WebAssembly/wasi-libc/pull/463
---
 .../swift_build_support/products/wasisysroot.py                 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/swift_build_support/swift_build_support/products/wasisysroot.py b/utils/swift_build_support/swift_build_support/products/wasisysroot.py
index 9f0c2a3a571..2f355760e2f 100644
--- a/utils/swift_build_support/swift_build_support/products/wasisysroot.py
+++ b/utils/swift_build_support/swift_build_support/products/wasisysroot.py
@@ -203,7 +203,7 @@ class WasmLLVMRuntimeLibs(cmake_product.CMakeProduct):
         self.cmake_options.define('LIBCXX_ENABLE_SHARED:BOOL', 'FALSE')
         self.cmake_options.define('LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY:BOOL', 'FALSE')
         self.cmake_options.define('LIBCXX_ENABLE_EXCEPTIONS:BOOL', 'FALSE')
-        self.cmake_options.define('LIBCXX_ENABLE_FILESYSTEM:BOOL', 'FALSE')
+        self.cmake_options.define('LIBCXX_ENABLE_FILESYSTEM:BOOL', 'TRUE')
         self.cmake_options.define('LIBCXX_CXX_ABI', 'libcxxabi')
         self.cmake_options.define('LIBCXX_HAS_MUSL_LIBC:BOOL', 'TRUE')
 
-- 
2.46.0

